### PR TITLE
[Forge] Add simple test for fast sync throughput.

### PR DESCRIPTION
**Pull request by JoshLind**
### Description
This PR adds a simple forge test that benchmarks the throughput of state sync when running fast sync. Specifically, it counts and displays the throughput of the number of state values synced per second.

Note: most of this PR is simply just refactoring code into reusable methods to avoid code duplication 😄 

### Test Plan
Ran this on forge and verified the results, e.g., a very quick run showed:

```
State sync throughput : 21837 state values / sec
Test Ok
```

- - -
This change is [Reviewable](https://reviewable.io/reviews/aptos-labs/aptos-core/5584)

### CODE REVIEW

**Review by rustielin — APPROVED** [review]
LGTM!

File: crates/transaction-emitter-lib/src/emitter/account_minter.rs [hunk 1]
```diff
@@ -121,7 +121,7 @@ impl<'t> AccountMinter<'t> {
                 coins_for_source
             );
 
-            if req.promt_before_spending {
```
Comment (rustielin) [inline crates/transaction-emitter-lib/src/emitter/account_minter.rs:124]: nice catch 😁

File: testsuite/testcases/src/state_sync_performance.rs [hunk 2]
```diff
@@ -22,73 +28,93 @@ impl Test for StateSyncFullnodePerformance {
...
-            .swarm()
-            .full_nodes()
-            .map(|v| v.peer_id())
-            .collect::<Vec<_>>();
-        if all_fullnodes.is_empty() {
-            return Err(anyhow::format_err!(
-                "Fullnode performance tests require at least 1 fullnode!"
-            ));
-        }
+        let all_fullnodes = get_fullnodes_and_check_setup(ctx, self.name())?;
```
Comment (igor-aptos) [inline testsuite/testcases/src/state_sync_performance.rs:31]: changes here https://github.com/aptos-labs/aptos-core/pull/5490/files#diff-a83aaea40a44dc1164c9cdc3bf5882913d55a755ba1fa82f15643a826e35dd50 might simplify this logic?

**Review by igor-aptos — APPROVED** [review]

File: testsuite/testcases/src/state_sync_performance.rs (hunk 2 above)
Comment (JoshLind) [reply testsuite/testcases/src/state_sync_performance.rs:31]: Nice, I'll clean it up when I do my next forge related task 😄

**Pull request merged**